tests/test_discovery.py

```diff
@@ -2331,5 +2340,129 @@ def test_get_media(self):
         self.assertEqual(b"standing in for media", response)
 
 
+if HAS_UNIVERSE:
+
+    class Universe(unittest.TestCase):
+        def test_validate_credentials_with_no_universe(self):
+            fake_universe = "foo.com"
+
+            http = google_auth_httplib2.AuthorizedHttp(
+                credentials=None, http=build_http()
+            )
+            discovery = read_datafile("zoo.json")
+            service = build_from_document(
+                discovery,
+                http=http,
+                client_options=google.api_core.client_options.ClientOptions(
+                    universe_domain=universe.DEFAULT_UNIVERSE
+                ),
+            )
+
+            assert service._validate_credentials()
+
+            # TODO(omairn): uncomment when universe configured universe is leveraged.
+            # http = google_auth_httplib2.AuthorizedHttp(credentials=None, http=build_http())
+            # discovery = read_datafile("zoo.json")
+            # service = build_from_document(discovery, http=http, client_options=google.api_core.client_options.ClientOptions(universe_domain=fake_universe))
+
+            # with self.assertRaises(universe.UniverseMismatchError):
+            #     service._validate_credentials()
+
+            # TODO(omairn): Add test case for not specifying universe domain via client option.
```

Comment:
Please remove commented code and add TODO comments linking to a Github issue in this repo

Reply:
addressed.